### ISSUE TITLE
TestBuildKanikoInsecureRegistry: waiting for pod to complete: condition error: pod already in terminal phase: Failed

### ISSUE THREAD

**Issue by dgageot** [GoogleContainerTools/skaffold#3302]
See https://source.cloud.google.com/results/invocations/2124e0bd-ac84-4a96-a767-f23e40f1afad/targets/runtimes-common%2Fskaffold%2Fpresubmit/log

**Comment by balopat** [comment]
we got another one today: 

```
--- FAIL: TestBuildKanikoInsecureRegistry (49.56s)
    build_test.go:374: skaffold build: exit status 1, Generating tags...
         - kaniko-insecure -> 34.70.129.71:5000/kaniko-insecure:v1.0.0-124-gceb1a88
        Checking cache...
         - kaniko-insecure: Not found. Building
        Creating kaniko secret [default/e2esecret]...
        Building [kaniko-insecure]...
        Storing build context at /tmp/context-06fa688fd5a564d4c05a6136236a8f8c.tar.gz
    panic.go:406: time="2019-12-02T23:58:37Z" level=fatal msg="build failed: build failed: building [kaniko-insecure]: kaniko build for [kaniko-insecure]: waiting for pod to complete: condition error: pod already in terminal phase: Failed"
```

**Comment by balopat** [comment]
```
 FAIL: TestBuildKanikoInsecureRegistry (100.63s)
    build_test.go:374: skaffold build: exit status 1, Generating tags...
         - kaniko-insecure -> 35.192.142.24:5000/kaniko-insecure:v1.0.0-126-g1e7eae7
        Checking cache...
         - kaniko-insecure: Not found. Building
        Creating kaniko secret [default/e2esecret]...
        Building [kaniko-insecure]...
        Storing build context at /tmp/context-927db788ff5827bed280887b3668b752.tar.gz
        ERROR: logging before flag.Parse: E1203 00:58:48.857940       1 metadata.go:142] while reading 'google-dockercfg' metadata: http status code: 404 while fetching url http://metadata.google.internal./computeMetadata/v1/instance/attributes/google-dockercfg
        ERROR: logging before flag.Parse: E1203 00:58:48.864515       1 metadata.go:159] while reading 'google-dockercfg-url' metadata: http status code: 404 while fetching url http://metadata.google.internal./computeMetadata/v1/instance/attributes/google-dockercfg-url
        error checking push permissions -- make sure you entered the correct tag name, and that you are authenticated correctly, and try again: checking push permission for "35.192.142.24:5000/kaniko-insecure:v1.0.0-126-g1e7eae7": creating push check transport for 35.192.142.24:5000 failed: Get http://35.192.142.24:5000/v2/: dial tcp 35.192.142.24:5000: connect: connection refused
    panic.go:406: time="2019-12-03T00:58:51Z" level=fatal msg="build failed: build failed: building [kaniko-insecure]: kaniko build for [kaniko-insecure]: waiting for pod to complete: condition error: pod already in terminal phase: Failed"
```

**Comment by tejal29** [comment]
@balopat i remember you had a PR for a flaky test around CC summit? Is this the same test? Is this addressed now?

**Comment by balopat** [comment]
I think we can close this now, it is not happening anymore, let's close it -- it might have been a kokoro specific thing...

**Comment by balopat** [comment]
Still happening :( 

```
--- FAIL: TestBuildKanikoInsecureRegistry (182.35s)
    build_cluster_test.go:50: skaffold build: exit status 1, Generating tags...
         - kaniko-insecure -> 35.223.218.95:5000/kaniko-insecure:v1.2.0-185-g434c40c
        Checking cache...
         - kaniko-insecure: Not found. Building
        Creating kaniko secret [default/e2esecret]...
        Building [kaniko-insecure]...
        ERROR: logging before flag.Parse: E0129 05:50:07.949404       1 metadata.go:142] while reading 'google-dockercfg' metadata: http status code: 404 while fetching url http://metadata.google.internal./computeMetadata/v1/instance/attributes/google-dockercfg
        ERROR: logging before flag.Parse: E0129 05:50:07.952548       1 metadata.go:159] while reading 'google-dockercfg-url' metadata: http status code: 404 while fetching url http://metadata.google.internal./computeMetadata/v1/instance/attributes/google-dockercfg-url
        INFO[0000] Resolved base name golang:1.12.9-alpine3.10 to golang:1.12.9-alpine3.10
        INFO[0000] Resolved base name alpine:3.10 to alpine:3.10
        INFO[0000] Resolved base name golang:1.12.9-alpine3.10 to golang:1.12.9-alpine3.10
        INFO[0000] Resolved base name alpine:3.10 to alpine:3.10
        INFO[0000] Downloading base image golang:1.12.9-alpine3.10
        INFO[0000] Error while retrieving image from cache: getting file info: stat /cache/sha256:39f9d94d45e404c98c3aa3c9b2406185f46ffddefeb45cdd9b778024626302fc: no such file or directory
        INFO[0000] Downloading base image golang:1.12.9-alpine3.10
        INFO[0001] Downloading base image alpine:3.10
        INFO[0001] Error while retrieving image from cache: getting file info: stat /cache/sha256:de78803598bc4c940fc4591d412bffe488205d5d953f94751c6308deeaaa7eb8: no such file or directory
        INFO[0001] Downloading base image alpine:3.10
        INFO[0001] Built cross stage deps: map[0:[/app]]
        INFO[0001] Downloading base image golang:1.12.9-alpine3.10
        INFO[0002] Error while retrieving image from cache: getting file info: stat /cache/sha256:39f9d94d45e404c98c3aa3c9b2406185f46ffddefeb45cdd9b778024626302fc: no such file or directory
        INFO[0002] Downloading base image golang:1.12.9-alpine3.10
        INFO[0002] Using files from context: [/kaniko/buildcontext/main.go]
        INFO[0002] Checking for cached layer 35.223.218.95:5000/kaniko-insecure/cache:7639513a00fda7c1711399172c73bd2158870198a72d7b4450141a24f3d6533f...
        INFO[0002] No cached layer found for cmd COPY main.go .
        INFO[0002] Unpacking rootfs as cmd COPY main.go . requires it.
        INFO[0010] Taking snapshot of full filesystem...
        INFO[0015] Using files from context: [/kaniko/buildcontext/main.go]
        INFO[0015] COPY main.go .
        INFO[0015] Taking snapshot of files...
        INFO[0015] RUN go build -o /app main.go
        INFO[0015] cmd: /bin/sh
        INFO[0015] args: [-c go build -o /app main.go]
        INFO[0015] Pushing layer 35.223.218.95:5000/kaniko-insecure/cache:7639513a00fda7c1711399172c73bd2158870198a72d7b4450141a24f3d6533f to cache now
        INFO[0015] Taking snapshot of full filesystem...
        INFO[0016] Pushing layer 35.223.218.95:5000/kaniko-insecure/cache:04c4a70afcc5cfa9b4e8d4c455f4032379d0ba778fd975e289b8cd49375a821d to cache now
        INFO[0016] Saving file /app for later use.
        INFO[0016] Deleting filesystem...
        INFO[0017] Downloading base image alpine:3.10
        INFO[0017] Error while retrieving image from cache: getting file info: stat /cache/sha256:de78803598bc4c940fc4591d412bffe488205d5d953f94751c6308deeaaa7eb8: no such file or directory
        INFO[0017] Downloading base image alpine:3.10
        INFO[0018] Checking for cached layer 35.223.218.95:5000/kaniko-insecure/cache:bd5ba88067551119d57eb4ced79318e89aed0d80f54293d05bbca165803ae85f...
        INFO[0018] No cached layer found for cmd COPY --from=builder /app .
        INFO[0018] Unpacking rootfs as cmd COPY --from=builder /app . requires it.
        INFO[0018] Taking snapshot of full filesystem...
        INFO[0018] CMD ["./app"]
        INFO[0018] No files changed in this command, skipping snapshotting.
        INFO[0018] COPY --from=builder /app .
        INFO[0018] Taking snapshot of files...
        INFO[0018] Pushing layer 35.223.218.95:5000/kaniko-insecure/cache:bd5ba88067551119d57eb4ced79318e89aed0d80f54293d05bbca165803ae85f to cache now
        error pushing image: failed to push to destination 35.223.218.95:5000/kaniko-insecure:v1.2.0-185-g434c40c: Head http://35.223.218.95:5000/v2/kaniko-insecure/blobs/sha256:c2fc8184f3b19c218ed0c5120e211391de3f0de948a70e327e29b5c39a860300: read tcp 10.32.5.187:40934->35.223.218.95:5000: read: connection reset by peer
    panic.go:563: time="2020-01-29T05:52:19Z" level=fatal msg="build failed: building [kaniko-insecure]: waiting for pod to complete: condition error: pod already in terminal phase: Failed"

```
